### PR TITLE
feat: support empty user-agent

### PR DESCRIPTION
**Pull request by Sn0rt**
set the user-agent with the user's value when the user set the user-agent by header.

If the user does not set the `user-agent` field, and the `use_default_user_agent` is also false, then the current library will not set this field.

If the user does not configure the `user-agent` field, and the `use_default_user_agent`  is true, then the current field is set it with the default value.

### CODE REVIEW

**Comment by soulbird** [comment]
It is best not to change the default behavior

**Comment by Sn0rt** [comment]
> It is best not to change the default behavior

thx. fixed.

File: lib/resty/http.lua [hunk 1]
```diff
@@ -725,7 +726,7 @@ function _M.send_request(self, params)
             headers["Host"] = self.host
         end
     end
-    if not headers["User-Agent"] then
+    if use_default_user_agent ~= false and not headers["User-Agent"] then
         headers["User-Agent"] = _M._USER_AGENT
```
Comment (soulbird) [inline lib/resty/http.lua:730]: `use_default_user_agent ` maybe `nil`

File: t/19-user-agent-header.t [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+location = /a {
+    content_by_lua_block {
+        local httpc = require("resty.http").new()
+        assert(httpc:connect("127.0.0.1", ngx.var.server_port),
+            "connect should return positively")
+
+        local res, err = httpc:request{
+            path = "/b",
+            headers = {
+                ["X_Foo"] = "bar",
```
Comment (leslie-tsang) [inline t/19-user-agent-header.t:198]: Need a test case to cover empty headers.

File: lib/resty/http.lua (hunk 1 above)
Comment (Sn0rt) [reply lib/resty/http.lua:730]: > `use_default_user_agent ` maybe `nil`

I think this code can be triggered when use_default_user_agent is `nil` or `true`

File: t/19-user-agent-header.t (hunk 2 above)
Comment (Sn0rt) [reply t/19-user-agent-header.t:198]: add thx.

**Review by juststillthinking — APPROVED** [review]

**Review by soulbird — APPROVED** [review]
